src/main/java/com/coveros/selenified/Capabilities.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    /**
+     * returns the classes defined all browser details
+     *
+     * @return Browser
+     */
+    public Browser getBrowser() {
+        return browser;
+    }
+
+    public void setInstance(int instance) {
+        this.instance = instance;
+    }
+
+    /**
+     * Retrieves the instance of the test running. This references the invocation count from TestNG, allowing looping,
+     * and specifies which test run this is
+     *
+     * @return Integer
+     */
+    public int getInstance() {
+        return instance;
+    }
+
+    /**
+     * Obtains the set system values for the proxy, and adds them to the desired
+     * desiredCapabilities
+     */
+
+    public void setupProxy() {
+        // are we running through a proxy
+        if (System.getProperty(PROXY_INPUT) != null) {
+            // set the proxy information
+            Proxy proxy = new Proxy();
+            proxy.setHttpProxy(System.getProperty(PROXY_INPUT));
+            desiredCapabilities.setCapability(CapabilityType.PROXY, proxy);
+        }
+    }
+
+    /**
+     * Sets the device desiredCapabilities based on the browser selection
+     *
+     * @throws InvalidBrowserException If a browser that is not one specified in the
+     *                                 Selenium.Browser class is used, this exception will be thrown
+     */
+    public void setupBrowserCapability() throws InvalidBrowserException {
+        if (System.getProperty("hub") != null) {
+            switch (browser.getName()) { // check the browser
+                case HTMLUNIT:
+                    desiredCapabilities = DesiredCapabilities.htmlUnit();
+                    break;
+                case FIREFOX:
+                    desiredCapabilities = DesiredCapabilities.firefox();
+                    break;
+                case CHROME:
+                    desiredCapabilities = DesiredCapabilities.chrome();
+                    break;
+                case INTERNETEXPLORER:
+                    desiredCapabilities = DesiredCapabilities.internetExplorer();
+                    break;
+                case EDGE:
+                    desiredCapabilities = DesiredCapabilities.edge();
+                    break;
+                case ANDROID:
+                    desiredCapabilities = DesiredCapabilities.android();
+                    break;
+                case IPHONE:
+                    desiredCapabilities = DesiredCapabilities.iphone();
+                    break;
+                case IPAD:
+                    desiredCapabilities = DesiredCapabilities.ipad();
+                    break;
+                case SAFARI:
+                    desiredCapabilities = DesiredCapabilities.safari();
+                    break;
+                case OPERA:
+                    desiredCapabilities = DesiredCapabilities.operaBlink();
+                    break;
+                case PHANTOMJS:
+                    desiredCapabilities = DesiredCapabilities.phantomjs();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "phantomjs"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)